complaint_search/es_interface.py

```diff
@@ -21,10 +22,125 @@ def get_es():
             timeout=100)
     return _ES_INSTANCE
 
-def search(format="json"):
+def _create_and_append_bool_should_clauses(es_field_name, value_list, 
+    filter_list, with_subitems=False, es_subitem_field_name=None):
+    if value_list:
+        if not with_subitems:
+            term_list = [ {"terms": {es_field_name: [value]}} 
+                for value in value_list ]
+            filter_list.append({"bool": {"should": term_list}})
+        else:
+            item_dict = defaultdict(list)
+            for v in value_list:
+                # -*- coding: utf-8 -*-
+                v_pair = v.split(u'\u2022')
+                print v_pair
+                # No subitem
+                if len(v_pair) == 1:
+                    # This will initialize empty list for item if not in item_dict yet
+                    item_dict[v_pair[0]]
+                elif len(v_pair) == 2:
+                    # put subproduct into list
+                    item_dict[v_pair[0]].append(v_pair[1])
+
+            print "item_dict", item_dict
+            # Go through item_dict to create filters
+            f_list = []
+            for item, subitems in item_dict.iteritems():
+                item_term = {"terms": {es_field_name: [item]}}
+                # Item without any subitems
+                if not subitems:
+                    f_list.append(item_term)
+                else:
+                    subitem_term = {"terms": {es_subitem_field_name: subitems}}
+                    f_list.append({"and": {"filters": [item_term, subitem_term]}})
+
+            filter_list.append({"bool": {"should": f_list}})
+
+def search(fmt="json", field="what_happened", size=10, frm=0, 
+    sort=None, search_term=None, min_date=None, max_date=None, 
+    company=None, product=None, subproduct=None, issue=None, subissue=None, 
+    state=None, consumer_disputed=None, company_response=None):
+
```

Comment:
That's a lot of arguments!

I would think you could use `**kwargs` except for the values you actually want to default -or-

```python
defaults = {...}
composed = defaults.update(**kwargs)
```